apps/chat_websocket/lib/chat_websocket/channels/room_channel.ex

```diff
@@ -130,6 +132,41 @@ defmodule ChatWebsocket.RoomChannel do
     end
   end
 
+  def handle_in("room:archive", _params, socket) do
+    %{room: room, user: user} = socket.assigns
+
+    with {:ok, record} <- UpdateRoom.call(room.id, user, "archived"),
+         :ok <- broadcast!(socket, "room:archived", record),
+         {:ok, _} <- maybe_update_support_presence(record),
+         subscriptions <- ListSubscriptions.call(record),
+         :ok <- broadcast_user_subscriptions!(subscriptions) do
+      reply(:ok, %{room: record}, socket)
+    else
+      _ -> reply(:error, "Error archiving room", socket)
+    end
+  end
+
+  defp maybe_update_support_presence(room) do
+    case room.type do
+      "support" -> TrackSupportRooms.update(room, %{state: room.state})
+      _ -> {:ok, true}
+    end
+  end
```

Comment:
The compiler is going to issue a warning that function signatures should be grouped together. In other words, all the `handle_in` definitions should follow one another. To fix this, private functions like `maybe_update_support_presence` should come after the last `handle_in` definition.

Though compiler warnings like these don't seem like a big deal, it's good to resolve them whenever possible. Otherwise important compiler warnings that are a big deal will be much harder to spot amongst the noise.